analyzers/src/SonarAnalyzer.CFG/ShimLayer/LightupHelpers.cs

```diff
@@ -53,20 +55,25 @@ internal static bool CanWrapObject(object obj, Type underlyingType)
             return false;
         }
 
-        ConcurrentDictionary<Type, bool> wrappedObject = SupportedObjectWrappers.GetOrAdd(underlyingType, _ => new ConcurrentDictionary<Type, bool>());
-
         // Avoid creating the delegate if the value already exists
-        bool canCast;
-        if (!wrappedObject.TryGetValue(obj.GetType(), out canCast))
+        if (!SupportedObjectWrappers.TryGetValue(underlyingType, out var wrappedObject))
```

Comment:
is there a reason not to make these changes upstream, and afterwards copy them here? 

The initial idea of the ShimLayer was that we only keep the code from them. Modifying it may require changes in licensing, keeping track of the changes and mentioning them.

Reply:
We are quite a lot out of sync with the two SourceGenrators already. I don't know about the other files, as it is unclear what we added and what was copied from StyleCop. This file is out of sync as well (we have 877 lines and StyleCop 678).
Waiting for the upstream repo to react is also an unfortunate burden. I can create a PR, but I would keep it separate from this one. Once the upstream PR is merged, we may copy the file content over, but it seems we added some functionality in this file in the past.

Reply:
yes, we should not wait for upstream but we should always contribute back, and it doesn't seem we have this process in place

Reply:
I created DotNetAnalyzers/StyleCopAnalyzers/pull/3711